src/components/pages/register.tsx

```diff
@@ -1,16 +1,130 @@
-import { SelfServiceRegistrationFlow } from "@ory/kratos-client"
-import { useState } from "react"
+import { AxiosError } from "axios"
+import {
+  SelfServiceRegistrationFlow,
+  SubmitSelfServiceRegistrationFlowBody,
+} from "@ory/kratos-client"
+import { history } from "../../store/history"
+import { useState, InputHTMLAttributes, useEffect } from "react"
+import { KratosSdk, handleFlowError } from "../../kratos"
+import config from "../../store/config"
+import { Flow } from "../kratos"
 
 interface RegisterProps {
   flowData?: KratosFlowData
 }
 
 const Register = ({ flowData: flowDataProp }: RegisterProps) => {
-  const [flowData] = useState<SelfServiceRegistrationFlow | undefined>(
+  const [flowData, setFlowData] = useState<SelfServiceRegistrationFlow | undefined>(
     flowDataProp?.registrationData,
   )
+  useEffect(() => {
+    console.log("use effect")
+    if (flowData) {
+      return
+    }
 
-  return <div className="register">{JSON.stringify(flowData)}</div>
+    console.log("no flow data")
+    const kratos = KratosSdk(config.kratosBrowserUrl)
+    const params = new URLSearchParams(window.location.search)
+    const flowId = params.get("flow")
+
+    // flow id exists, we can fetch the flow data
+    if (flowId) {
+      console.log("fetching flow data")
+      kratos
+        .getSelfServiceRegistrationFlow(String(flowId),
+          undefined,
+        {withCredentials: true}
+        )
+        .then(({ data }) => {
+          setFlowData(data)
+        })
+        .catch(
+          handleFlowError({
+            history,
+            resetFlow: () => {
+              console.log("reset flow with id")
+              setFlowData(undefined)
+              history.push(`/register/email`)
+            },
+          }),
+        )
+      return
+    }
+
+    console.log("no flow id")
+    // need to initialize the flow
+    kratos
+      .initializeSelfServiceRegistrationFlowForBrowsers(
+        params.get("return_to") || undefined,
+        {withCredentials: true}
+      )
+      .then(({ data }) => {
+        setFlowData(data)
+      })
+      .catch(
+        handleFlowError({
+          history,
+          resetFlow: () => {
+            console.log("reset flow with id")
+            setFlowData(undefined)
+            console.log("set as undefined")
+            history.push("/register/email")
+          },
+        }),
+      )
+  }, [flowData])
+
+  const onSubmit = async (values: SubmitSelfServiceRegistrationFlowBody) => {
+    const kratos = KratosSdk(config.kratosBrowserUrl)
+    history.push(`/register/email?flow=${flowData?.id}`)
```

Comment:
I just realised perhaps this line is the problem. We need to add `?flow=${flowData.id}` to the url but still pick up the state that gets added at the comment down below. Haven't tried it just guessing.

Reply:
Perhaps we need to use the second `state` argument to `history.push` or something like that